tools/walletextension/wallet_extension.go

```diff
@@ -141,54 +136,36 @@ func (we *WalletExtension) handleHTTPEthJSON(resp http.ResponseWriter, req *http
 		return
 	}
 
-	// We unmarshall the JSON request.
-	var reqJSONMap map[string]interface{}
+	// We unmarshall the JSON request
+	var reqJSONMap map[string]json.RawMessage
 	err = json.Unmarshal(body, &reqJSONMap)
 	if err != nil {
 		logAndSendErr(resp, fmt.Sprintf("could not unmarshall JSON-RPC request body to JSON: %s. "+
 			"If you're trying to generate a viewing key, visit %s", err, pathViewingKeys))
 		return
 	}
-	method := reqJSONMap[reqJSONKeyMethod]
-	fmt.Printf("Received %s request from wallet: %s\n", method, body)
-
-	// We encrypt the request's params with the enclave's public key if it's a sensitive request.
-	maybeEncryptedBody, err := we.encryptParamsIfNeeded(body, method, reqJSONMap)
-	if err != nil {
-		logAndSendErr(resp, fmt.Sprintf("could not encrypt request parameters: %s", err))
-		return
-	}
 
-	// We forward the request on to the Obscuro node.
-	nodeResp, err := forwardMsgOverWebsocket(websocketProtocol+we.hostAddr, maybeEncryptedBody)
+	var method string
+	err = json.Unmarshal(reqJSONMap[reqJSONKeyMethod], &method)
```

Comment:
Do we need all this wacky unmarshalling? Can't we just do, `method := reqJSONMap[reqJSONKeyMethod]`?